src/components/AutoCompleteInput.tsx

```diff
@@ -0,0 +1,208 @@
+import {
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverBody,
+  useColorModeValue,
+} from "@chakra-ui/react";
+import { Box, Text } from "@chakra-ui/react";
+import React, { useCallback, useEffect } from "react";
+import { useRef, useState } from "react";
+
+type AutoCompleteProps = {
+  children: React.ReactNode;
+  triggerRef: React.RefObject<HTMLTextAreaElement>;
```

Comment:
We should probably allow Input elements as well
```suggestion
  triggerRef: React.RefObject<HTMLTextAreaElement> | React.RefObject<HTMLInputElement>;
```